lib/delegate.ml

```diff
@@ -34,7 +34,10 @@ let publish_distrib ?token ?distrib_uri ~dry_run ~msg ~archive ~yes ~draft pkg =
       Pkg.tag pkg >>= fun version ->
       (match distrib_uri with
       | Some uri -> Ok uri
-      | None -> Pkg.infer_distrib_uri pkg)
+      | None ->
+          (* This is an absolute non-sense but we're keeping it to preserve the
+             behaviour until 2.0 and removal of delegates *)
+          Pkg.infer_github_distrib_uri pkg)
```

Comment:
That would be a slight behavior change but we could require a `--dist-uri` when using a delegate, the user would get a more precise error message.

Reply:
Yeah I think it would make sense but it should probably be done in a separate PR, what do you think?

Reply:
Yes it's probably better indeed